File: Seori/Weekly Contest 413/3274. Check if Two Chessboard Squares Have the Same Color.py
class Solution:
    def checkTwoChessboards(self, coordinate1: str, coordinate2: str) -> bool:
        # 좌표의 앞뒤 합이 짝수이면 검정, 홀수이면 흰색이다. 따라서 두 좌표의 합이 짝수이면 같은 색이다.
        # If the sum of the coordinates is even, it is black, and if it is odd, it is white. Therefore, if the sum of two coordinates is even, they are the same color.
        return not (ord(coordinate1[0]) + int(coordinate1[1]) + ord(coordinate2[0]) + int(coordinate2[1])) % 2

File: Lucy/Weekly Contest 413/3274_Check_if_Two_Chessboard_Squares_Have_the_Same_Color.js
/**
 * 3274. Check if Two Chessboard Squares Have the Same Color
 *
 * url: https://leetcode.com/problems/check-if-two-chessboard-squares-have-the-same-color/
 * topic: Math, String
 * difficulty: Easy
 * date: 2024.09.10(TUE)
 */

/**
 * @param {string} coordinate1
 * @param {string} coordinate2
 * @return {boolean}
 */
var checkTwoChessboards = function (coordinate1, coordinate2) {
  const getSquareColor = ([x, y]) => {
    const isEven = (coordinateY) => coordinateY % 2 === 0;
    const coordinateMap = new Map(
      Object.entries({
        a: { odd: "black", even: "white" },
        b: { odd: "white", even: "black" },
        c: { odd: "black", even: "white" },
        d: { odd: "white", even: "black" },
        e: { odd: "black", even: "white" },
        f: { odd: "white", even: "black" },
        g: { odd: "black", even: "white" },
        h: { odd: "white", even: "black" },
      })
    );
    return isEven(y) ? coordinateMap.get(x).even : coordinateMap.get(x).odd;
  };

  return (
    getSquareColor(coordinate1.split("")) ===
    getSquareColor(coordinate2.split(""))
  );
};

console.log(checkTwoChessboards("a1", "c3"));
console.log(checkTwoChessboards("a1", "h3"));
